core/engine/src/value/inner/nan_boxed.rs

```diff
@@ -0,0 +1,864 @@
+//! This `[JsValue]` inner type is a NaN-boxed value, which is a 64-bits value
+//! that can represent any JavaScript value. If the integer is a non-NaN value,
+//! it will be stored as a 64-bits float. If it is a `f64::NAN` value, it will
+//! be stored as a quiet `NaN` value. Subnormal numbers are regular float.
+//!
+//! For any other type of values, the value will be stored as a 51-bits non-zero
+//! integer.
+//!
+//! In short, the memory layout of a NaN-boxed value is as follows:
+//!
+//! | Type of           | Bit Layout | Comment |
```

Comment:
The table layout doesn't seem right in this comment, there's 3 headings but 4 columns underneath, do they properly match up?

Reply:
There's only 3 columns in the comment. You might be confused by the pointer hexa explanation that uses `\|` to escape the pipe sign.

Here's a rendered version:

<img width="1800" alt="CleanShot 2024-12-31 at 18 53 56@2x" src="https://github.com/user-attachments/assets/fe186a07-e935-4d4e-ad28-9f8368cf2daa" />